src/config/accounts.c

```diff
@@ -844,8 +844,6 @@ accounts_set_last_status(const char* const account_name, const char* const value
     if (accounts_account_exists(account_name)) {
         if (value) {
             g_key_file_set_string(accounts, account_name, "presence.laststatus", value);
-        } else {
-            g_key_file_remove_key(accounts, account_name, "presence.laststatus", NULL);
```

Comment:
I'm not really in this part of the code so I might ask some stupid questions.
Here is the first one: why remove this?

Reply:
Well that is the tricky part and a reason why I thought that maybe setting status messages should be refactored. 
I am not sure why that part is there, but my guess would be for overwriting an older status message like: `/status set online` would overwrite an older setting `/status set away eating`. But currently using `/status set online` automatically implies `/status set online online`.
The reason why it was removed is that this function is called without a value before we try to restore the last status which will remove the last status message from our config file.

Reply:
Seems like the function is for auto away: https://github.com/profanity-im/profanity/commit/54e225aa5330b7a3c20bda38264e5ad0cfdfcdc5
Probably so that something like "away (ready to talk)" does not happen. But I don't get why that should remove the setting from the file.